SDL_ttf.c

```diff
@@ -1449,6 +1548,68 @@ static SDL_Surface *Create_Surface_Blended(int width, int height, SDL_Color fg,
     return textbuf;
 }
 
+static SDL_Surface* Create_Surface_LCD(int width, int height, SDL_Color fg, SDL_Color bg, Uint32 *color)
+{
+    const int alignment = Get_Alignement() - 1;
+    SDL_Surface *textbuf = NULL;
+    Uint32 bgcolor;
+
+    /* Background color */
+    bgcolor = (bg.a << 24) | (bg.r << 16) | (bg.g << 8) | bg.b;
+
+    /* Underline/Strikethrough color style */
+    *color = (bg.a << 24) | (fg.r << 16) | (fg.g << 8) | fg.b;
+
+    /* Create the target surface if required */
+    if (width != 0) {
+        /* Create a surface with memory:
+         * - pitch is rounded to alignment
+         * - adress is aligned
+         */
+        Sint64 size;
+        void *pixels, *ptr;
+        /* Worse case at the end of line pulling 'alignment' extra blank pixels */
+        int pitch = (width + alignment) * 4;
+        pitch += alignment;
+        pitch &= ~alignment;
+        size = height * pitch + sizeof (void *) + alignment;
+        if (size < 0 || size > SDL_MAX_SINT32) {
+            /* Overflow... */
+            return NULL;
+        }
+
+        ptr = SDL_malloc((size_t)size);
+        if (ptr == NULL) {
+            return NULL;
+        }
+
+        /* address is aligned */
+        pixels = (void *)(((size_t)ptr + sizeof(void *) + alignment) & ~alignment);
+        ((void **)pixels)[-1] = ptr;
+
+        textbuf = SDL_CreateRGBSurfaceWithFormatFrom(pixels, width, height, 0, pitch, SDL_PIXELFORMAT_ARGB8888);
+        if (textbuf == NULL) {
+            SDL_free(ptr);
+            return NULL;
+        }
+
+        /* Let SDL handle the memory allocation */
+        textbuf->flags &= ~SDL_PREALLOC;
+        textbuf->flags |= SDL_SIMD_ALIGNED;
```

Comment:
This should only be set if the surface is allocated with SDL_SIMDAlloc(), otherwise it can cause a crash.

Reply:
Ah, but I see you abuse the way the SIMD memory allocation works so that this is done correctly. Maybe we should add a function `SDL_SIMDAllocAligned(len, align)`?